lib/internal/modules/esm/worker.js

```diff
@@ -0,0 +1,25 @@
+const { workerData } = require('internal/worker_threads');
+const { ESMLoader } = require('internal/modules/esm/loader');
+
+const { comsChannel } = workerData;
+const lock = new Int32Array(comsChannel, 0, 4); // Required by Atomics
+const requestResponseData = new Uint8Array(comsChannel, 4, 2044); // for TextEncoder/Decoder
+
+const publicESMLoader = new ESMLoader();
+
+Atomics.store(lock, 0, 0); // Send 'ready' signal to main
+Atomics.notify(lock, 0); // Notify main of signal
+
+while (true) { // event loop
+  Atomics.wait(lock, 0, 0); // This pauses the while loop
+  // Worker is now active and main is sleeping
+  const { data, type } = JSON.parse(
+    (new TextDecoder().decode(requestResponseData))
+      .replaceAll('\x00', '') // strip empty space (not a great solution)
+  );
+  const response = await publicESMLoader[type](data);
```

Comment:
this lacks error handling, I would expect the worker to somehow let the main thread via the proxy about the failure

Reply:
It's a draft because it's not done yet 😉 good shout though, thanks

Reply:
I'l give another peak when I see it moves to ready for review :)